pkg/products/monitoring/reconciler.go

```diff
@@ -457,19 +460,51 @@ func (r *Reconciler) reconcileScrapeConfigs(ctx context.Context, serverClient k8
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
-func (r *Reconciler) reconcileTemplates(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
-	// Iterate over template_list
-	for _, template := range r.Config.GetTemplateList() {
-		// create it
-		_, err := r.createResource(ctx, template, serverClient)
+func (r *Reconciler) reconcileDashboards(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+
+	for _, dashboard := range r.Config.GetDashboards() {
+		err := r.reconcileGrafanaDashboards(ctx, serverClient, dashboard)
 		if err != nil {
-			return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to create/update monitoring template %s: %w", template, err)
+			return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to create/update grafana dashboard %s: %w", dashboard, err)
 		}
-		r.Logger.Infof("Reconciling the monitoring template %s was successful", template)
+		r.Logger.Infof("Reconciling the grafana dashboard  %s was successful", dashboard)
 	}
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
+func (r *Reconciler) reconcileGrafanaDashboards(ctx context.Context, serverClient k8sclient.Client, dashboard string) (err error) {
+
+	grafanaDB := &grafanav1alpha1.GrafanaDashboard{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dashboard,
+			Namespace: r.Config.GetOperatorNamespace(),
+		},
+	}
+
+	opRes, err := controllerutil.CreateOrUpdate(ctx, serverClient, grafanaDB, func() error {
+		specJSON, name, err := getSpecDetailsForDashboard(dashboard)
+		if err != nil {
+			return err
+		}
```

Comment:
Minor/optional thing: this could maybe be moved to the top of the `reconcileGrafanaDashboards` function rather than here in the mutate function that's passed to `controllerutil.CreateOrUpdate`. Doing so would avoid the extra k8s client GET that `controllerutil.CreateOrUpdate` will do before running this mutate function (we know at a much earlier stage that an error may occur here, and we don't get any extra info by waiting until `controllerutil.CreateOrUpdate` calls the mutate function).

Reply:
This is the main portion of JSON definition that needs to be updated if there are any changes. From our code base, anything that isnt objectMeta, and that needs to be updates are usually placed inside the createorupdate(). This method "getSpecDetailsForDashboard" is not a client operation , but rather a static helper method that returns the JSON spec and "name" as strings.

Reply:
I think it could still be pulled out of the mutate function here, into the `reconcileGrafanaDashboards` function though: `specJSON` and `name` should still be in scope in here in the mutate function for when they're used in L494/L495.

Like I said though, it's no big deal, it's an optional change :slightly_smiling_face:

Reply:
I understood now your point. Agreed. And Fixed!